### PR TITLE
Fix Zephyr support

### PR DESCRIPTION
**Pull request by dgloeck**
These are the changes I would do to the Zephyr support code to make it work again and to bring it in shape for upstream merging. I would like the patches to be accepted here before I attempt to bring them upstream (zephyrproject-rtos/zephyr#2933).

@lpereira, I think it is best to squash your commit and mine into a single one for upstream. Yours is missing a Signed-off-by line with your name and eMail address. Can I add it?

### CODE REVIEW

File: src/rtos/Zephyr.c [hunk 1]
```diff
@@ -36,34 +39,86 @@ struct Zephyr_thread {
...
 };
 
 static struct Zephyr_params Zephyr_params_list[] = {
 	{
 		.target_name = "cortex_m",
 		.pointer_width = 4,
-		.reg_stacking = &rtos_standard_Cortex_M4F_FPU_stacking,
+		.reg_stacking = {
+			{
+				36,
```
Comment (galak) [inline src/rtos/Zephyr.c]: Can we add a comment or something about what this magic number is.

File: src/rtos/Zephyr.c [hunk 2]
```diff
@@ -36,34 +39,86 @@ struct Zephyr_thread {
...
+				-1,
+				ARRAY_SIZE(arm_callee_saved),
+				NULL,
+				arm_callee_saved,
+			}, {
+				32,
```
Comment (galak) [inline src/rtos/Zephyr.c]: Can we add a comment or something about what this magic number is.

**Comment by galak** [comment]
I merged the trivial patches into the branch.

**Comment by galak** [comment]
@dgloeck Do you mind adding a bit more details in the commit messages.

**Comment by dgloeck** [comment]
Will do ...tomorrow.

**Comment by dgloeck** [comment]
The meaning of the "magic numbers" is now clearer because of the use of C99 designated initializers.

All commits now have a more detailed commit message.

I have also changed the way we undo the stack realigning and the skip the floating point registers to use the rtos_Cortex_M_stack_align function provided by OpenOCD. There no longer is a separate commit for this as the information to correct the stack pointer is part of the rtos_register_stacking structures (which are now stored outside of the Zephyr_params structure, btw.).

**Pull request merged**